src/controller_base.cpp

```diff
@@ -230,11 +230,33 @@ void controller_base::handle_event(const SDL_Event& event)
 		break;
 
 	case SDL_MOUSEWHEEL:
-#if defined(_WIN32) || defined(__APPLE__)
-		mh_base.mouse_wheel(-event.wheel.x, event.wheel.y, is_browsing());
-#else
-		mh_base.mouse_wheel(event.wheel.x, event.wheel.y, is_browsing());
+		// Down and right are positive in Wesnoth's map.
+		// Up and right are positive in SDL_MouseWheelEvent on all
+		// platforms:
+		//     https://wiki.libsdl.org/SDL2/SDL_MouseWheelEvent
+		// Except right is wrongly negative on X11 in SDL < 2.0.18:
+		//     https://github.com/libsdl-org/SDL/pull/4700
+		//     https://github.com/libsdl-org/SDL/commit/515b7e9
+		// and on Wayland in SDL < 2.0.20:
+		//     https://github.com/libsdl-org/SDL/commit/3e1b3bc
+		// Fixes issue #7404, which is a regression caused by pull #2481
+		// that fixed issue #2218.
+		{
+			int xmul = 1;
+#if !defined(_WIN32) && !defined(__APPLE__)
+			const char* video_driver = SDL_GetCurrentVideoDriver();
```

Comment:
This this cheap enough to execute on any SDL_MOUSEWHEEL event? Probably "yes", but worth checking. You've probably ran it locally and didn't observe any noticeable slowdown, @pehjota ?

Reply:
UPD: yes, it's cheap enough. I've tested on 1.17 now and I'm not able to notice any slow-down. Thanks!

Reply:
Right, as mentioned in the commit message, I ran this locally with SDL 2.0.14 and 2.28.5.  No noticeable slowdown even on my 2.40-GHz Intel Core 2 Duo (GMA 4500MHD) laptop.

It's only two to four cheap function calls and a handful of assignments and comparisons on each SDL_MouseWheelEvent.  But it can be improved slightly by caching the version/driver checks result in a static variable.  Done, in PRs #8185 and #8186.

Thanks again!